### DISCUSSION TITLE
Bulk Requests - API Design for Long-Running Tasks

### DISCUSSION THREAD

**Discussion by thientrinhcoder** [QilinCloud/QilinCloud#5] [General]
A long-running task is an operation that requires a considerable amount of server resources and/or time. To avoid blocking the client, the task must be completed asynchronously without the persistent connection between the client and the server.

After submitting the task, the client needs to poll to a provided URL for the task-execution progress. If there is WebSockets support, the client can also be notified via WebSockets when execution finishes, instead of polling.

**Comment by marc-costea** [comment]
This provided URL can be specific just for this task, but I do not prefer this method.
Instead we should stay with an general endpoint for the asynchronous results and identify the specific result by some id, like the [X-Request-Id](https://github.com/QilinCloud/QilinCloud/wiki/Request-ID-&-Correlation-ID#x-request-id), because this is made for linking request and result.

**Comment by thientrinhcoder** [comment]
Let me make clear, your idea in "bulk request" is like an entry point that allows the external system can post/get multiple items at once time. but underlying, they still are split to every single object and then processed by every single API calling asynchronous with status.
If that is, we just overcome the rate limit, not overcome the quota limit, right?

**Comment by thientrinhcoder** [comment]
and could you describe for us about real use case? 
I'm figuring this feature is useful for bulk import data via API or webhook which will develop in next phase.
